src/Microsoft.Health.Fhir.SqlServer/Features/Storage/TvpRowGeneration/BulkDateTimeSearchParameterV2RowGenerator.cs

```diff
@@ -0,0 +1,48 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using Microsoft.Health.Fhir.Core.Features.Search;
+using Microsoft.Health.Fhir.Core.Features.Search.SearchValues;
+using Microsoft.Health.Fhir.SqlServer.Features.Schema.Model;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Storage.TvpRowGeneration
+{
+    internal class BulkDateTimeSearchParameterV2RowGenerator : BulkSearchParameterRowGenerator<DateTimeSearchValue, BulkDateTimeSearchParamTableTypeV2Row>
+    {
+        private short _lastUpdatedSearchParamId;
+
+        public BulkDateTimeSearchParameterV2RowGenerator(SqlServerFhirModel model, SearchParameterToSearchValueTypeMap searchParameterTypeMap)
+            : base(model, searchParameterTypeMap)
+        {
+        }
+
+        internal override bool TryGenerateRow(int offset, short searchParamId, DateTimeSearchValue searchValue, out BulkDateTimeSearchParamTableTypeV2Row row)
+        {
+            if (searchParamId == _lastUpdatedSearchParamId)
+            {
+                // this value is already stored on the Resource table.
+                row = default;
+                return false;
+            }
+
+            row = new BulkDateTimeSearchParamTableTypeV2Row(
+                offset,
+                searchParamId,
+                searchValue.Start,
+                searchValue.End,
+                (searchValue.Start - searchValue.End).Ticks > TimeSpan.TicksPerDay,
+                searchValue.IsMin,
+                searchValue.IsMax);
+
+            return true;
+        }
+
+        protected override void Initialize()
+        {
+            _lastUpdatedSearchParamId = Model.GetSearchParamId(SearchParameterNames.LastUpdatedUri);
```

Comment:
>Model.GetSearchParamId(SearchParameterNames.LastUpdatedUri) [](http://example.com/codeflow?start=40&length=59)

I can understand usage of Initialize() for something deffered, but we using Model object which we get in constructor.

Can this be part of constructor?

Reply:
Doesn't seem like it is possible to push this to constructor. The `GetSearchParamId` depends on the Model object being intialized and throws if moved to the ctor.